src/app/games/trexo/TrexoMinimax.ts

```diff
@@ -0,0 +1,46 @@
+import { BoardValue } from 'src/app/jscaip/BoardValue';
+import { Coord } from 'src/app/jscaip/Coord';
+import { MGPNode } from 'src/app/jscaip/MGPNode';
+import { Minimax } from 'src/app/jscaip/Minimax';
+import { Player, PlayerOrNone } from 'src/app/jscaip/Player';
+import { SCORE } from 'src/app/jscaip/SCORE';
+import { TrexoMove } from './TrexoMove';
+import { TrexoNode, TrexoRules } from './TrexoRules';
+import { TrexoState } from './TrexoState';
+
+export class TrexoMinimax extends Minimax<TrexoMove, TrexoState> {
+
+    public getListMoves(node: TrexoNode): TrexoMove[] {
+        return TrexoRules.get().getLegalMoves(node.gameState);
+    }
+    public getBoardValue(node: TrexoNode): BoardValue {
+        let score: number = 0;
+        const state: TrexoState = node.gameState;
+        const lastPlayer: Player = state.getCurrentOpponent();
+        let lastPlayerAligned5: boolean = false;
+        for (let x: number = 0; x < TrexoState.SIZE; x++) {
+            // for every column, starting from the bottom of each column
+            for (let y: number = 0; y < TrexoState.SIZE; y++) {
+                // while we haven't reached the top or an empty space
+                const pieceOwner: PlayerOrNone = state.getPieceAtXY(x, y).owner;
+                if (pieceOwner.isPlayer()) {
+                    const tmpScore: number = TrexoRules.getSquareScore(state, new Coord(x, y));
```

Comment:
hmm les rules ont une notion de score ? :/

Reply:
C'est en effet plus lié au Minimax, mais Rules utilise getVictoriousCoords pour savoir si y a une victoire, et getVictoriousCoords utilise les N-In-A-Row helper, qui ont en effet p-e plus leurs place en Minimax qu'en Rules, que faire ?

Reply:
utiliser les NInARowHelper dans TrexoMinimax également pour la partie score je dirais, non ?

Reply:
Alors j'ai de la duplication entre TrexoMinimax.getBoardValue qui retourne toutes les valeurs possible de plateau
puis le même code presque pour TrexoRules.getGameStatus qui les maps en GameStatus.
Mettre le getBoardValue dans Rules est moyen, mais faire utiliser le Minimax.getBoardValue faire de l'import cyclique donc nope.
Comment nommer la class qui s'occuperait getBoardValue :/ ?

Reply:
```
    public static getSquareScore(state: TrexoState, coord: Coord): number {
        return TrexoRules.TREXO_HELPER.getSquareScore(state, coord);
    }
```

Tu sais pas simplement appeler TREXO_HELPER.getSquareScore plutôt que TrexoRules.getSquareScore ? :p

Reply:
et ça n'irait pas simplement dans NInRowHelper le getBoardValue spécifique aux NInARow ?